keanu-project/src/main/java/io/improbable/keanu/model/regression/RegressionModelBuilder.java

```diff
@@ -0,0 +1,217 @@
+     *                       This will represent sigmas if no or ridge regularization is used and will represent betas if lasso regularization is used.
+     * @return An updated version of this builder
+     */
+    public RegressionModelBuilder setPriorOnWeightsAndIntercept(double mean, double scaleParameter) {
+        setPriorOnWeights(RegressionWeights.fillPriorOnWeights(this.inputTrainingData.getShape(), mean), RegressionWeights.fillPriorOnWeights(this.inputTrainingData.getShape(), scaleParameter));
+        setPriorOnIntercept(mean, scaleParameter);
+        return this;
+    }
+
+    /**
+     * Builds and fits LinearRegressionModel using the data passed to the builder.
+     * @return A linear regression model from the data passed to the builder
+     */
+    public RegressionModel<DoubleTensor> buildLinearRegressionModel() {
+        checkVariablesAreCorrectlyInitialised();
+
+        RegressionGraph<DoubleTensor> regressionGraph = new RegressionGraph<>(
+            this.inputTrainingData.getShape(),
+            RegressionModel.gaussianOutputTransform(observationSigma),
+            getInterceptVertex(),
+            getWeightsVertex()
+        );
+
+        performDataFitting(regressionGraph, outputLinearTrainingData);
+        return new RegressionModel<>(regressionGraph);
+    }
+
+    /**
+     * Builds and fits LogisticRegressionModel using the data passed to the builder.
+     * @return A linear regression model from the data passed to the builder
+     */
+    public RegressionModel<BooleanTensor> buildLogisticRegressionModel() {
+        checkVariablesAreCorrectlyInitialised();
+
+        RegressionGraph<BooleanTensor> regressionGraph = new RegressionGraph<>(
+                this.inputTrainingData.getShape(),
+                RegressionModel.logisticOutputTransform(),
+                getInterceptVertex(),
+                getWeightsVertex()
+        );
+
+        performDataFitting(regressionGraph, outputLogisticTrainingData);
+        return new RegressionModel<>(regressionGraph);
+    }
+
+    private void checkVariablesAreCorrectlyInitialised() {
+        if (inputTrainingData == null || (outputLogisticTrainingData == null && outputLinearTrainingData == null)) {
+            throw new IllegalArgumentException("You have not provided both the input and output variables");
+        }
+        if (priorOnWeightsMeans == null || priorOnWeightsScaleParameters == null) {
+            setPriorOnWeights(RegressionWeights.fillPriorOnWeights(this.inputTrainingData.getShape(), DEFAULT_MU), RegressionWeights.fillPriorOnWeights(this.inputTrainingData.getShape(), DEFAULT_SCALE_PARAMETER));
+        }
+
+        if (priorOnInterceptMean == null || priorOnInterceptScaleParameter == null) {
+            setPriorOnIntercept(DEFAULT_MU, DEFAULT_SCALE_PARAMETER);
+        }
+    }
+
+    private DoubleVertex getInterceptVertex() {
+        switch (this.regularization) {
```

Comment:
The trouble with enums is you end up with ugly `switch` statements. If you use a class hierarchy instead then you can do it all with method dispatch. By adding methods to your enum you can get the best of both worlds. Then your method would just be:

    private DoubleVertex getInterceptVertex() {
        return this.regularization.getInterceptVertex(priorOnInterceptMean, priorOnInterceptScaleParameter)
    }